components/script/dom/mutationrecord.rs

```diff
@@ -18,6 +21,46 @@ pub struct MutationRecord {
 
     //property for target node
     target: JS<Node>,
+
+    //property for attribute name
+    attribute_name: Option<DOMString>,
+
+    //property for attribute namespace
+    attribute_namespace: Option<DOMString>,
+
+    //property for old value
+    old_value: Option<DOMString>,
+}
+
+impl MutationRecord {
+    pub fn new(record_type: DOMString, target: &Node) -> Root<MutationRecord> {
+        let boxed_record = box MutationRecord::new_inherited(record_type, target);
+        return reflect_dom_object(boxed_record, window_from_node(target).deref(), MutationRecordBinding::Wrap);
+    }
+
+    fn new_inherited(record_type: DOMString, target: &Node) -> MutationRecord {
+        MutationRecord {
+            reflector_: Reflector::new(),
+            record_type: record_type,
+            target: JS::from_ref(target),
+            attribute_name: Default::default(),
+            attribute_namespace: Default::default(),
+            old_value: Default::default(),
+        }
+    }
+
+    // setter for attr_name
+    pub fn SetAttributeName(&mut self, attr_name: DOMString) {
+        self.attribute_name = Some(attr_name);
+    }
+    // setter for attr_namespace
+    pub fn SetAttributeNamespace(&mut self, attr_namespace: DOMString) {
+        self.attribute_namespace = Some(attr_namespace);
+    }
+    // setter for attr_oldvalue
+    pub fn SetoldValue(&mut self, attr_oldvalue: DOMString) {
```

Comment:
These should not be `&mut self`; instead, use `&self` and interior mutability by storing `DOMRefCell<Option<DOMString>>` and use `*self.attribute_name.borrow_mut() = ...`.